ts/client/services/rpc.ts

```diff
@@ -25,31 +25,46 @@ export class RPC {
         this.server = server;
     }
 
-    static async init(
-        port: number,
-        storageManager: StorageManager,
-        pool: TransferPool
-    ): Promise<RPC> {
+    static async init(api: CoreAPI, port: number): Promise<RPC> {
+        const { l2Storage, depositPool } = api;
+
         const server = fastify({ logger: true });
-        server.setErrorHandler(console.log);
+        server.setErrorHandler(console.error);
 
-        server.register(async function(fastify: FastifyInstance) {
+        server.register(async (fastify: FastifyInstance) => {
             fastify.get("/", helloHandler);
-            fastify.get("/user/state/:stateID", async function(
-                request: any,
-                reply: any
-            ) {
-                const stateID = request.params.stateID;
-                const state = await storageManager.state.get(stateID);
-                return state.toJSON();
-            });
-            fastify.post("/tx", tx, async function(request: any, reply: any) {
-                const bytes = arrayify(request.body.bytes);
-                const transfer = TransferOffchainTx.deserialize(bytes);
-                console.log(transfer.toString());
-                pool.push(transfer);
-                return { txHash: transfer.hash() };
-            });
+            fastify.get<{ Params: { stateID: number } }>(
+                "/user/state/:stateID",
+                async function(request) {
+                    const stateID = request.params.stateID;
+                    const state = await l2Storage.state.get(stateID);
+                    return state.toJSON();
+                }
+            );
+            fastify.post<{ Body: { bytes: string } }>(
+                "/tx",
+                tx,
+                async request => {
+                    const bytes = arrayify(request.body.bytes);
+                    const transfer = TransferOffchainTx.deserialize(bytes);
+                    console.log(transfer.toString());
+                    depositPool.pushDeposit(transfer.message());
```

Comment:
Should this be the transferPool instead of depositPool?

Reply:
Yes, my mistake. Will correct.